src/widgets/vitals/vitals-overview.component.tsx

```diff
@@ -111,7 +111,7 @@ const VitalsOverview: React.FC<VitalsOverviewProps> = () => {
     });
 
   const launchVitalsBiometricsForm = () => {
-    const url = `/patient/${patientUuid}/vitalsbiometrics/form`;
+    const url = "/patient/:patientUuid/vitalsbiometrics/form";
```

Comment:
This is an invalid URL.

Should not be the path template, but the actual path.

Reply:
Hello, @FlorianRappl, 
Just clarifying:
path like : "${openmrsSpaBase}/patient/:patientUuid/vitalsbiometrics/form"

Reply:
No.

```ts
`/patient/${patientUuid}/vitalsbiometrics/form`
```

is correct. In your screenshot also all vitals are empty, i.e., no valid patient selected. Make sure the patientUuid is defined and valid though.

Reply:
Hello @FlorianRappl,
> Make sure the patientUuid is defined and valid

![image](https://user-images.githubusercontent.com/61249902/111942686-ce720200-8af9-11eb-9b98-019d563fc603.png)

I think it's defined correctly here. Because I made a few code changes to check for its value:

![image](https://user-images.githubusercontent.com/61249902/111943227-02015c00-8afb-11eb-8b06-ac4bf0cfb4b5.png)



> In your screenshot also all vitals are empty, i.e., no valid patient selected.

This page is to add the vitals and Biometrics. 

I tried to load the form with the help of the patientUuid from the chart/summary URL. The same page gets loaded.

![image](https://user-images.githubusercontent.com/61249902/111942253-f57c0400-8af8-11eb-820c-43203c87b375.png)

And I was able to add the vitals to the patient as well through the page that gets rendered.

![image](https://user-images.githubusercontent.com/61249902/111942446-4c81d900-8af9-11eb-9645-ff8dee2e9fd8.png)

Do have a look, please.
Thank you.

Reply:
Are you making the requested changes now?

Because I don't see any push. Still it's using the invalid format.

Reply:
@FlorianRappl,
I am working on it.
`/patient/${patientUuid}/vitalsbiometrics/form`
This URL is not working, even though the `patientUuid` is correct. 
I just wanted to bring to your notice that the above changes that I made are working fine. 
I will soon commit the changes as per the mentioned guideline.
Thank you.

Reply:
I'm not sure here.

Because we use the same approach in the drug ordering and it works. Also there is not any part that does such replacement that I am aware of.

> I will soon commit the changes as per the mentioned guideline.

What changes? I guess there may be a misunderstanding here. You mean the requested changes?

Reply:
@FlorianRappl,
When we click on the "Add" button on the vitals widget, it takes us to an empty page.

![vitals](https://user-images.githubusercontent.com/61249902/112020998-da3fe180-8b56-11eb-8ea9-c02cc309002a.gif)

For this only the issue [MF-499](https://issues.openmrs.org/browse/MF-499), is created.
And I am trying to work on the link.
Do tell me if I am missing something here.
Thank you.

Reply:
Clicking on the plus button triggers (among other things):

```
https://openmrs-spa.org/openmrs/ws/rest/v1/visit?patient=null&v=custom:(uuid,encounters:(uuid,encounterDatetime,form:(uuid,name),location:ref,encounterType:ref,encounterProviders:(uuid,display,provider:(uuid,display))),patient:(uuid,uuid),visitType:(uuid,name,display),attributes:(uuid,display,value),location:(uuid,name,display),startDatetime,stopDatetime)
```

Note that `patient` is `null`. This problem is **not** exclusive to vitals, but can also be seen on other (visit-related) parts. I guess it may have to do with usage of React Router instead of vanilla JS URL properties. But that is just a guess. Actually finding the root cause and solving this is what should be done here.

Reply:
Note that the problem above is presumably not the root cause, but just some side-effect of a bigger issue. Most requests are fine and contain the valid patient UUID - I guess there is some issue in the code with the response.

But again, just observations. The bottom line is that there is no issue with the current URL, but rather with the handling of the response / the component itself.

Reply:
@FlorianRappl,
What do you suggest should be the further steps to correct this?